webpack/move_to_foreman/common/helpers.js

```diff
@@ -3,19 +3,31 @@ import { addToast } from 'foremanReact/redux/actions/toasts';
 const urlBuilder = (controller, action, id = undefined) =>
   `/${controller}/${id ? `${id}/` : ''}${action}`;
 
-const urlWithSearch = (base, searchQuery) =>
-  `/${base}?search=${searchQuery}`;
+const urlWithSearch = (base, searchQuery) => `/${base}?search=${searchQuery}`;
 
 const stringIsInteger = (value) => {
   // checking for positive integers only
   const reg = new RegExp('^[0-9]+$');
   return reg.test(value);
 };
 
+const getErrorMessegeByResponseMessage = (message) => {
+  const errorMessageHash = {
+    '404 Not Found': 'The subscription cannot be found upstream',
+    '410 Gone': 'The subscription is no longer available',
+  };
+  return errorMessageHash[message];
+};
+
 export const getResponseErrorMsgs = ({ data }) => {
   if (data) {
-    const messages = (data.errors || data.displayMessage || data.message || data.error.message);
-    return (Array.isArray(messages) ? messages : [messages]);
+    const messages =
+      getErrorMessegeByResponseMessage(data.displayMessage) ||
```

Comment:
This helper is supposed to be generic and will most likely be placed at more places in future, not just on the subscriptions page. Therefore I'd like to avoid wiring subscription specific messages in. I'm not sure yet what is the best way to enable custom messages in the `apiError` helper :-/. Maybe injecting the function?